pennylane/ftqc/lattice.py

```diff
@@ -0,0 +1,189 @@
+        nx.relabel_nodes(self._graph, mapping, copy=False)
+
+    def set_node_attributes(self, attribute_name: str, attributes: Dict):
+        r"""Add attributes to the nodes of the Networkx graph.
+
+        Args:
+            attribute_name (str): Name of the node attribute to set.
+            attributes (Dict): A dict with node labels as keys and attributes as values.
+        """
+        # TODO: This method could be renamed later as it's possible that this method is only for stablizers setup.
+        nx.set_node_attributes(self._graph, attributes, attribute_name)
+
+    def get_node_attributes(self, attribute_name: str):
+        r"""Return node attributes.
+        Args:
+            attribute_name (str): Name of the node attribute
+        """
+        return nx.get_node_attributes(self._graph, attribute_name)
+
+    def set_edge_attributes(self, attribute_name: str, attributes: Union[str, Dict]):
+        r"""Add attributes to the edges of the Network graph.
+
+        Args:
+            attribute_name (str): Name of the edge attribute to set.
+            attributes (Dict): Edge attributes to set. It accepts a dict with node labels as keys and attributes as values or a scalar to set the new attribute of egdes with.
+        """
+        # TODO: This method could be renamed later as it's possible that this method is only for the entanglement setup.
+        nx.set_edge_attributes(self._graph, attributes, attribute_name)
+
+    def get_edge_attributes(self, attribute_name: str):
+        r"""Add attributes to the edges of the Network graph.
+
+        Args:
+            attribute_name (str): Name of the edge attribute to set.
+        """
+        # TODO: This method could be renamed later as it's possible that this method is only for the entanglement setup.
+        return nx.get_edge_attributes(self._graph, attribute_name)
+
+    def get_neighbors(self, node):
+        r"""Returns the neighbors of a given node in the lattice.
+
+        Args:
+            node: a target node label.
+        """
+        return self._graph.neighbors(node)
+
+    def get_nodes(self):
+        r"""Returns all nodes in the lattice."""
+        return self._graph.nodes
+
+    def get_edges(self):
+        r"""Returns all edges in the lattice."""
+        return self._graph.edges
+
+    def get_graph(self) -> nx.Graph:
+        r"""Returns the underlying NetworkX graph object representing the lattice."""
+        return self._graph
+
+
+def generate_lattice(lattice, dims: List[int]) -> Lattice:
+    r"""Generates a :class:`~pennylane.ftqc.Lattice` object for a given lattice shape and dimensions.
+
+    Args:
+        lattice (str): Shape of the lattice. Input values can be ``'chain'``, ``'rectangle'``, ``'honeycomb'``, ``'triangle'``, ``'cubic'``.
+        dims(List[int]): Number of nodes in each direction of the lattice.
+
+    Returns:
+        a :class: `~pennylane.ftqc.Lattice` object.
```

Comment:
Might want to double check that this renders correctly, but I think there's no space between `:class:` and the first backtick.

```suggestion
        a :class:`~pennylane.ftqc.Lattice` object.
```

Reply:
Nice catch